src/lib/server-side-props/get-challenge.ts

```diff
@@ -0,0 +1,15 @@
+import { ChallengeDetailsBySlugDocument, ChallengeDetailsBySlugQuery, ChallengeDetailsBySlugQueryResult } from "app/generated/graphql";
+import { getApolloClient } from "app/lib/apollo-client/rsc";
+
+export type Challenge = ChallengeDetailsBySlugQueryResult["data"]["challenge"];
+
+export async function getChallenge(slug: string): Promise<Challenge> {
+  const client = getApolloClient();
+
+  const response = await client.query<ChallengeDetailsBySlugQuery>({
```

Comment:
MA `client.query` accepts a second type parameter for the query arguments. That type is also generated by graphql codegen as `...QueryArgs`.